### DISCUSSION TITLE
How do I effectively add multiple items to one page?

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#165] [Alfred Q&A]
I am planning on inserting a questionnaire into my experiment. I want multiple items to appear on the same page. Each items should be followed by a single-choice bar. How can I elegantly add several items without needing too much code?

**Comment by Adsecularis** [comment]
Ah, that was a wonderful little problem to crack. And I found one answer utilizing a for-loop.

After the hook I placed a for-loop, in the example I set the range to 3, so everything inside the for-loop will be executed 3 times.
And each time it will first include the Text and then the SingleChoiceBar.
The text is static and the same for each iteration of the loop.
Same with the possible answers in the SingleChoiceBar, only that they are smuggled in each time from a list in my content.py by thesmuggler.
Because SingleChoiceBars need unique names I used a formal string (f" ") and included {i+1} so it includes the i from the range and adds 1 so it is 1,2,3 instead of 0,1,2.

You could use the range to display multiple texts, by using something like path=f"files/text_{i}.md" to load a different markdown file each time.
Or use static answers for your SingleChoiceBar by typing them out. Like ("Yes", "No", "Maybe", name=f"scale_{i+1}")
Of course the loop can have whatever range you desire.
```python
import alfred3 as al
from thesmuggler import smuggle
content = smuggle("files/content.py")

exp = al.Experiment()

@exp.member
class ItemPage(al.Page):
    
    def on_first_show(self):
        for i in range (3):
            self += al.Text("Your Question", align="center")
            self += al.SingleChoiceBar(
                *content.choices,
                name=f"scale_{i+1}",
            )

if __name__ == "__main__":
    exp.run()
```

**Reply by Adsecularis** [reply]
And now I saw a simliar issue has already been solved in a similar way over here in #95
